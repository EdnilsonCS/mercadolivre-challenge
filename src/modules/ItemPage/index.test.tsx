import { composeStories } from '@storybook/testing-react';

import {
  render, screen,
} from '@testing-library/react';
import * as stories from './index.stories';

const {
  ItemPageDefault,
} = composeStories(stories);

jest.mock('next/image', () => ({
  __esModule: true,
  default: (props: any) => {
    return <img {...props} alt={props.alt} />;
  },
}));

describe('Test for search input component', () => {
  it('should match snapshot in default style', () => {
    const { container } = render(<ItemPageDefault />);

    expect(container).toMatchSnapshot();
  });

  it('should image render a url link', () => {
    render(<ItemPageDefault />);
    const findImage = screen.getByAltText('Title Value');

    expect(findImage).toHaveAttribute('src', '/Logo_ML.png');
  });

  it('should have a quantity and conditions information', () => {
    render(<ItemPageDefault />);

    const findText = screen.getByText('new - 240 vendidos');

    expect(findText).toBeTruthy();
  });

  it('should have a title', () => {
    render(<ItemPageDefault />);

    const findTitle = screen.getByText('Title Value');
    expect(findTitle).toBeTruthy();
  });

  it('should have a description', () => {
    render(<ItemPageDefault />);

    const findTitle = screen.getByText('description');
    expect(findTitle).toBeTruthy();
  });
});
