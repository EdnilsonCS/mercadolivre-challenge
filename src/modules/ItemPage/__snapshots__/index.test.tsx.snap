// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for search input component should match snapshot in default style 1`] = `
<div>
  <div>
    <div>
      <div>
        <img
          alt="Title Value"
          height="680px"
          src="/Logo_ML.png"
          width="680px"
        />
      </div>
      <div>
        <span>
          new
           
          -
           
          240
           
          vendidos
        </span>
        <h2>
          Title Value
        </h2>
        <span>
          value text
        </span>
        <button
          type="button"
        >
          Comprar
        </button>
      </div>
    </div>
    <div>
      <span>
        Descripci√≥n del producto
      </span>
      <p>
        description
      </p>
    </div>
  </div>
</div>
`;
